lib/select.h

```diff
@@ -96,10 +96,21 @@ int tpf_select_libcurl(int maxfds, fd_set* reads, fd_set* writes,
 
 /* Winsock and TPF sockets are not in range [0..FD_SETSIZE-1], which
    unfortunately makes it impossible for us to easily check if they're valid
+
+   With Winsock the valid range is between 0 and INVALID_SOCKET-1 according to
+   https://docs.microsoft.com/en-us/windows/win32/winsock/socket-data-type-2
```

Comment:
Extra comment unnecessary and I would change existing comment to mention only TPF.

Reply:
I think pointing to the Microsoft documentation makes sense here, because it seems like many people assume that SOCKETs on Windows can be any value (like a HANDLE) while there is indeed a definition available.